FreeRTOS/FreeRTOS-Plus-TCP
# Fix MISRA 2012 issues

Description
-----------
This PR fixes various (63) MISRA - 2012 violations detected by coverity:

- MISRA C-2012 Rule 8.2
    - Declaration uses a different type than "size_t uxStreamBufferMidSpace(const StreamBuffer_t *)".  [`uxStreamBufferMidSpace`]
    - Declaration uses a different type than "void vStreamBufferMoveMid(StreamBuffer_t *const, const size_t)" [`vStreamBufferMoveMid`]
- MISRA C-2012 Rule 10.4
    - Essential type of the left hand operand "254U" (unsigned) is not the same as that of the right operand "1"(signed). [x2] [`pxNew_AddrInfo`]
    - Implicit conversion of "xNDCache[x].ucAge" from essential type "unsigned 8-bit int" to different or narrower essential type "signed 32-bit int". [`vNDRefreshCacheEntry`]
    - Essential type of the left hand operand "xStreamBufferLessThenEqual(pxBuffer, pxBuffer->uxFront, uxNextHead)" (signed) is not the same as that of the right operand "(BaseType_t)0"(boolean).  [`uxStreamBufferAdd`]
    - Essential type of the left hand operand "xPeek" (signed) is not the same as that of the right operand "(BaseType_t)0"(boolean). [`uxStreamBufferGet`]
    - Essential type of the left hand operand "xSize > 4294967295U - xBytesRequiredForAlignment" (boolean) is not the same as that of the right operand "0"(signed). [BufferAllocation_2.c]
- MISRA C-2012 Rule 12.1
    - Missing parentheses on sub-expression of the operator.
        - `STATIC_ASSERT( ipconfigETHERNET_MINIMUM_PACKET_BYTES <= baMINIMAL_BUFFER_SIZE );`
- MISRA C-2012 Rule 18.4
- MISRA C-2012 Directive 4.14

Test Steps
-----------
Run coverity 2023

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
